### PR TITLE
Check for existence of network files before trying to update them

### PR DESCRIPTION
**Pull request by soapy1**
fixes https://github.com/hashicorp/vagrant/issues/11875 and https://github.com/hashicorp/vagrant/issues/11873

### CODE REVIEW

File: plugins/guests/redhat/cap/change_host_name.rb [hunk 1]
```diff
@@ -15,16 +15,20 @@ def self.change_host_name(machine, name)
             basename = name.split('.', 2)[0]
             comm.sudo <<-EOH.gsub(/^ {14}/, '')
               # Update sysconfig
-              sed -i 's/\\(HOSTNAME=\\).*/\\1#{name}/' /etc/sysconfig/network
+              if ls /etc/sysconfig | grep network; then
```
Comment (chrisroberts) [inline plugins/guests/redhat/cap/change_host_name.rb:18]: we can do a file check with:

`if [ -f /etc/sysconfig/network ]; then`

Comment (ikari7789) [reply plugins/guests/redhat/cap/change_host_name.rb:18]: Also safer as grepping on the result of `ls` would match any file with "network" in it's name inside of the folder leading to possible false positives.

**Review by chrisroberts — APPROVED** [review]
:+1: looks good

**Comment by brad2014** [comment]
@soapy1, @chrisroberts, I suspect this is still not quite right, my notes [here](https://github.com/hashicorp/vagrant/commit/7370f24318a4e3ced61922886f164834c7728f7e#r42356848).

File: plugins/guests/redhat/cap/change_host_name.rb [hunk 2]
```diff
@@ -15,9 +15,13 @@ def self.change_host_name(machine, name)
             basename = name.split('.', 2)[0]
             comm.sudo <<-EOH.gsub(/^ {14}/, '')
               # Update sysconfig
-              sed -i 's/\\(HOSTNAME=\\).*/\\1#{name}/' /etc/sysconfig/network
+              if [ -f /etc/sysconfig/network ]; then
+                sed -i 's/\\(HOSTNAME=\\).*/\\1#{name}/' /etc/sysconfig/network
+              fi
               # Update DNS
-              sed -i 's/\\(DHCP_HOSTNAME=\\).*/\\1\"#{basename}\"/' /etc/sysconfig/network-scripts/ifcfg-*
+              find /etc/sysconfig/network-scripts -maxdepth 1 -type f -name 'ifcfg-*' | xargs sed -i 's/\\(DHCP_HOSTNAME=\\).*/\\1\"#{basename}\"/'
```
Comment (brad2014) [inline plugins/guests/redhat/cap/change_host_name.rb:22]: This probably needs an xargs -r flag.

**Comment by soapy1** [comment]
@brad2014 thanks for following up on this!
Here is what I found, looking into it a little more. At first, it seems like this PR fixes the issue since the hostname is successfully updated without any errors. However, this is due to the addition of this line https://github.com/hashicorp/vagrant/pull/11877/files#diff-fd473b135e71706459b47b51cb86e249R24 which has a successful output. Since `comm.sudo` is only checking the return code of the last command this appears to works. Removing that line exposes the failure of `find /etc/sysconfig/network-scripts -maxdepth 1 -type f -name 'ifcfg-*' | xargs sed -i 's/\\(DHCP_HOSTNAME=\\).*/\\1\"#{basename}\"/'` which has a return code of `123`. Adding the `-r` argument to `xargs` fixes this. Here is a PR for this https://github.com/hashicorp/vagrant/pull/11910. 
Thanks again for your contribution ❤️

**Pull request merged**